libraries/bot-azure/src/main/java/com/microsoft/bot/azure/blobs/BlobsTranscriptStore.java

```diff
@@ -0,0 +1,423 @@
+                        conversations.add(conversation);
+                        if (conversations.size() == pageSize) {
+                            break;
+                        }
+                    }
+                }
+            }
+        } while (!StringUtils.isBlank(token) && conversations.size() < pageSize);
+
+        PagedResult<TranscriptInfo> pagedResult = new PagedResult<TranscriptInfo>() {
+            {
+                setItems(conversations);
+            }
+        };
+
+        if (pagedResult.getItems().size() == pageSize) {
+            pagedResult.setContinuationToken(pagedResult
+                .getItems().stream().reduce((first, second) -> second).get().getId());
+        }
+
+        return CompletableFuture.completedFuture(pagedResult);
+    }
+
+    public CompletableFuture<Void> deleteTranscript(String channelId, String conversationId) {
+        if (StringUtils.isBlank(channelId)) {
+            throw new IllegalArgumentException("channelId should not be null");
+        }
+
+        if (StringUtils.isBlank(conversationId)) {
+            throw new IllegalArgumentException("conversationId should not be null");
+        }
+
+        String token = null;
+        do {
+            String prefix = String.format("%1$s/%2$s/", sanitizeKey(channelId), sanitizeKey(conversationId));
+            Iterable<PagedResponse<BlobItem>> resultSegment = containerClient
+                .listBlobsByHierarchy(prefix).iterableByPage(token);
+            token = null;
+            for (PagedResponse<BlobItem> blobPage: resultSegment) {
+                for (BlobItem blobItem : blobPage.getValue()) {
+                    BlobClient blobClient = containerClient.getBlobClient(blobItem.getName());
+                    if (blobClient.exists()) {
+                        blobClient.delete();
+                    }
+
+                    // Get the continuation token and loop until it is empty.
+                    token = blobPage.getContinuationToken();
+                }
+            }
+        } while (!StringUtils.isBlank(token));
+
+        return CompletableFuture.completedFuture(null);
+    }
+
+    private CompletableFuture<Pair<Activity, BlobClient>> innerReadBlob(Activity activity) {
+        int i = 0;
+        while (true) {
+            try {
+                String token = null;
+                do {
+                    String prefix = String.format("%1$s/%2$s/",
+                        sanitizeKey(activity.getChannelId()), sanitizeKey(activity.getConversation().getId()));
+                    Iterable<PagedResponse<BlobItem>> resultSegment = containerClient
+                        .listBlobsByHierarchy(prefix).iterableByPage(token);
+                    token = null;
+                    for (PagedResponse<BlobItem> blobPage: resultSegment) {
+                        for (BlobItem blobItem : blobPage.getValue()) {
+                            BlobClient blobClient = containerClient.getBlobClient(blobItem.getName());
```

Comment:
Have you missed this [validation](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Azure.Blobs/BlobsTranscriptStore.cs#L342)?